### PR TITLE
feat: Storing of computed Contingency matrix

### PR DESCRIPTION
**Pull request by arjanoop**

### CODE REVIEW

**Review by MichaelRoeder — CHANGES_REQUESTED** [review]
The implementation is already good, but storing the data should be more generic, i.e, this solution explicitly only stores contingency matrices while the database could store any BLOB. This should be also implemented in this way so that we can use the same table / implementation for other results with BLOB data in the future.

File: src/main/resources/spring/database/schema/experiment-schema.sql [hunk 1]
```diff
@@ -50,3 +50,8 @@ subTaskId int NOT NULL,
 PRIMARY KEY (taskId, subTaskId)
 );
 
+CREATE TABLE IF NOT EXISTS ExperimentTasks_ContingencyMatrix (
```
[the remark this answers is not in the document]
Comment (arjanoop) [reply src/main/resources/spring/database/schema/experiment-schema.sql]: Refactored the code as per suggestion. 

- Updated DB Name as `ExperimentTasks_AdditionalBlobResults`
- Added field `resultType VARCHAR(20)` for generic reusability of DB Table.

File: src/main/java/org/aksw/gerbil/database/ExperimentDAOImpl.java [hunk 2]
```diff
@@ -16,7 +16,9 @@
  */
 package org.aksw.gerbil.database;
 
+import java.io.ByteArrayOutputStream;
 import java.io.IOException;
+import java.io.ObjectOutputStream;
```
Comment (MichaelRoeder) [inline src/main/java/org/aksw/gerbil/database/ExperimentDAOImpl.java:21]: I think this isn't needed, anymore. Can you please double check which of the inputs is actually necessary?

File: src/main/resources/spring/database/schema/experiment-schema.sql [hunk 3]
```diff
@@ -50,3 +50,9 @@ subTaskId int NOT NULL,
 PRIMARY KEY (taskId, subTaskId)
 );
 
+CREATE TABLE IF NOT EXISTS ExperimentTasks_AdditionalBlobResults (
+    taskId int NOT NULL,
+    resultType VARCHAR(20),
```
Comment (MichaelRoeder) [inline src/main/resources/spring/database/schema/experiment-schema.sql:55]: Please have a look at the `ExperimentTasks_AdditionalResults` table. The `resultType` should be a `resultId` and it should also be part of the primary key.

**Review by MichaelRoeder — CHANGES_REQUESTED** [review]
We are getting closer :smile: 

Can you please have a look at the usage of the "AdditionaResults" table? It has a resultId instead of the resultType that you suggest. The reason is that for a DB, handling numbers is much easier. Internally, GERBIL translates the IDs into the "meaning" of the result. For the additional results, this is done outside of the DB class. However, for the matrix, you can hard code the ID=1 --> contigency matrix for now.

File: src/main/java/org/aksw/gerbil/database/ExperimentDAOImpl.java (hunk 2 above)
Comment (arjanoop) [reply src/main/java/org/aksw/gerbil/database/ExperimentDAOImpl.java]: Removed unnecessary Imports.

File: src/main/resources/spring/database/schema/experiment-schema.sql (hunk 3 above)
Comment (arjanoop) [reply src/main/resources/spring/database/schema/experiment-schema.sql]: Refactored the implementation as per Suggestions.

- replaced `resultType` with `resultId`
- added `resultId` to primary key
- added `Enum ExerimentTaskBlobResultType` for getting resultId

**Review by MichaelRoeder — APPROVED** [review]
LGTM

**Pull request merged**